app/ide-desktop/lib/client/src/authentication.ts

```diff
@@ -0,0 +1,140 @@
+ * - Define a listener for {@link ipc.channel.openDeepLink} events (c.f., `preload.ts`).
+ *
+ * Then when the user clicks on a deep link from an external source to the IDE:
+ * - The OS redirects the user to the application.
+ * - The application emits an Electron {@link OPEN_URL_EVENT}.
+ * - The {@link OPEN_URL_EVENT} listener checks if the {@link URL} is a deep link.
+ * - If the {@link URL} is a deep link, the {@link OPEN_URL_EVENT} listener prevents Electron from
+ * handling the event.
+ * - The {@link OPEN_URL_EVENT} listener then emits an {@link ipc.channel.openDeepLink} event.
+ * - The {@link ipc.channel.openDeepLink} listener registered by the dashboard receives the event.
+ * Then it parses the {@link URL} from the event's {@link URL} argument. Then it uses the
+ * {@link URL} to redirect the user to the dashboard, to the page specified in the {@link URL}'s
+ * `pathname`. */
+import * as electron from 'electron'
+import * as ipc from 'ipc'
+import * as shared from '../shared'
+import opener from 'opener'
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/** Name of the Electron event that is emitted when a URL is opened in Electron (e.g., when the user
+ * clicks a link in the dashboard). */
+const OPEN_URL_EVENT = 'open-url'
+
+
+
+// ========================================
+// === Initialize Authentication Module ===
+// ========================================
+
+/** Configures all the functionality that must be set up in the Electron app to support
+ * authentication-related flows. Must be called in the Electron app `whenReady` event. */
+export const initAuthenticationModule = (
+    window: () => electron.BrowserWindow | null,
+) => {
+    initIpc()
+    initOpenUrlListener(window)
+}
+
+/** Registers an Inter-Process Communication (IPC) channel between the Electron application and the
+ * served website.
+ *
+ * This channel listens for {@link ipc.channel.openUrlInSystemBrowser} events. When this kind of
+ * event is fired, this listener will assume that the first and only argument of the event is a URL.
+ * This listener will then attempt to open the URL in a cross-platform way. The intent is to open
+ * the URL in the system browser.
+ *
+ * This functionality is necessary because we don't want to run the OAuth flow in the app. Users
+ * don't trust Electron apps to handle their credentials. */
+const initIpc = () => {
+    electron.ipcMain.on(ipc.channel.openUrlInSystemBrowser, (_event, url) => opener(url))
+}
+
+/** Registers a listener that fires a callback for `open-url` events, when the URL is a deep link.
+ *
+ * This listener is used to open a page in *this* application window, when the user is
+ * redirected to a URL with a protocol supported by this application.
+ *
+ * All URLs that aren't deep links (i.e., URLs that don't use the {@link shared.DEEP_LINK_SCHEME}
+ * protocol) will be ignored by this handler. */
```

Comment:
I was commenting it in my previous PR review - ignoring non deep links can cause REALLY long debugging process if something goes wrong. We should never ignore bad things. They should always be reported as warnings or errors.

Reply:
*This* handler is only interested in deep links. So by ignoring the non deep links, they end up having the default behaviour, which is to let Electron itself handle them. Is that not what we want? We're not disabling any functionality, only adding a special case for deep links.

Reply:
Do you know any nice way of providing here a logger instance to allow for logging?

Reply:
@indiv0 we should not allow Electron to handle any redirections, as it could potentially be dangerous for users. So, we should allow redirection of deep links and prevent any other redirection while logging an error. 

@PabloBuchu `client/src/index.ts` has logger access: `const logger = content.logger` - so you should be able to access it as well. Does it solve the issue?